controller.go

```diff
@@ -825,29 +825,41 @@ func parseAllocateConstraintsResponse(source interface{}, machine *machine) (Con
 	valid := coerced.(map[string]interface{})
 	constraintsMap := valid["constraints_by_type"].(map[string]interface{})
 	result := ConstraintMatches{
-		Interfaces: make(map[string]Interface),
-		Storage:    make(map[string]BlockDevice),
+		Interfaces: make(map[string][]Interface),
+		Storage:    make(map[string][]BlockDevice),
 	}
 
 	if interfaceMatches, found := constraintsMap["interfaces"]; found {
-		for label, value := range interfaceMatches.(map[string]interface{}) {
-			id := value.(int)
-			iface := machine.Interface(id)
-			if iface == nil {
-				return empty, NewDeserializationError("constraint match interface %q: %d does not match an interface for the machine", label, id)
+		matchMap := interfaceMatches.(map[string]interface{})
```

Comment:
This is really clunky because of all the extra casting needed - is there a better way of doing this?